src/typing_extensions.py

```diff
@@ -2810,6 +2811,41 @@ def get_protocol_members(tp: type, /) -> typing.FrozenSet[str]:
         return frozenset(_get_protocol_attrs(tp))
 
 
+if hasattr(typing, "Doc"):
+    Doc = typing.Doc
+else:
+    class Doc:
+        """Define the documentation of a type annotation using ``Annotated``, to be
+         used in class attributes, function and method parameters, return values,
+         and variables.
+
+        The value should be a positional-only string literal to allow static tools
+        like editors and documentation generators to use it.
+
+        This complements docstrings.
+
+        The string value passed is available in the attribute ``documentation``.
+
+        Example::
+
+            >>> from typing_extensions import Annotated, Doc
+            >>> def hi(to: Annotated[str, Doc("Who to say hi to")]) -> None: ...
+        """
+        def __init__(self, /, documentation: str) -> None:
```

Comment:
In the docs you say that this should be positional-only, which means the slash has to come after the argument:

```suggestion
        def __init__(self, documentation: str, /) -> None:
```

Reply:
🤦